docs/README.md

```diff
@@ -1846,6 +1862,188 @@ boost nuke -e production
+
+The only thing you need to do to deploy a whole new completely-independent copy of your application is to use a different name.
+
+## Debugging and testing Booster applications
+
+
+### Running Booster applications locally
+
+Having the ability to run your Booster application locally will dramatically reduce the deployment time of your application and speed-up the feedback loop. Furthermore, it allows developers to view application logs in the terminal.
+
+#### Prerequisites
+
+In order to run locally, it is necessary to add the `framework-provider-local` package to the project, and also define a new configuration environment `local`.
+
+First of all, we will have to install the local provider package as a new application dependency
+
+`npm install @boostercloud/framework-provider-local`
+
+Once the local provider is installed, we will also need to update our existing config, that can be found in `src/config/config.ts` in order to include the local environment configuration. More instructions on how to do add a new environment can be found [here](#Environments).
+
+#### Starting application
+
+This step is very simple, you only need to open a terminal window and run:
+
+```bash
+boost start -e local
+```
+
+By default, the application will be available at `http://localhost:3000`
+
+optionally, you could also specify what port you want the application to be running in but adding as arguments `-p <port-number>`
+
+```bash
+boost start -e local -p 3333
+```
+
+After a few seconds the Booster application should be ready to be hit by a client at `http://localhost:<port-number>`
+
+There will be two different endpoints available for our application:
+- `/auth`
+- `/graphql`
+
+#### Performing Auth requests
+Booster also provides you with user management for free, allowing you to sign-up, confirm and sign-in users. An example of a sign up would be as follow:
+
+`POST http://localhost:3000/auth/sign-up`
+
+```json
+{
+	"username": "user@test.com",
+	"password": "passw0rd!",
+	"userAttributes": {
+		"roles": ['User']
+	}
+}
+```
+
+#### Performing GraphQL requests
+
+We should now be able to perform queries and mutations to our GraphQL endpoint `http://localhost:<port-number>/graphql` with a client or tool, for example, a React App, Postwoman or Postman.
+
+`POST http://localhost:3000/graphql`
+
+Query
+```graphql
+mutation ChangeCartItem($cartId: ID!, $productId: ID!, $quantity: Float) {
+    ChangeCartItem(input: { cartId: $cartId, productId: $productId, quantity: $quantity })
+}
+```
+
+Variables
+```json
+{
+    "cartId": "e46d1d0e-5e7f-4934-850c-9559dc55af79",
+    "productId": "9214d0a9-0915-417d-852b-de79e54d8e95",
+    "quantity": 10
+}
+```
+
+#### Inspect Database information
+
+The databases for the local provider are just json files in the `<project-root>/.booster` folder. If you are wondering what data is available in the application you will only need to chose what file to look into.
```

Comment:
If there are some tools to deal with these files, it would be nice to explain here how to do simple queries or how to check the state in the "tables" from that files to help people debug the state.

Reply:
good point. I will look into what tools would help you search through json files

Reply:
@javiertoledo there are some interesting CLI tools to deal with the JSON files that nedb generates, but I don't think we should be the ones recommending those since they involve some learning. In any case, any JSON pretty-formatter online tool would help them previsualize data very easily